### PR TITLE
Switch to CxxWrap 0.10

### PR DESCRIPTION
**Pull request by fingolfin**
This upgrades to [CxxWrap 0.10.0 (release notes here)](https://github.com/JuliaInterop/CxxWrap.jl/issues/223). Of course this should only be merged in close coordination with `Polymake.jl`, in particular @saschatimme and @kalmarek ; see also https://github.com/oscar-system/Polymake.jl/pull/238

Whether we want to merge this now or later I don't know, but I wanted to test it now to see if everything works with CxxWrap 0.10.0; also, whether it has any effect on the crashes with see with the Julia nightly build...

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "Singular"
 uuid = "bcd08a7b-43d2-5ff7-b6d4-c458787f915c"
-version = "0.2.2"
+version = "0.2.3"
```
Comment (fingolfin) [inline Project.toml:3]: Perhaps this should be 0.3.0 as there is a binary incompatibility due to the CxxWrap switch? Honestly, though, I don't think it matters in practice for our purposes *shrug*

**Comment by fingolfin** [comment]
OK, so we also need to require Julia >= 1.3 if we want to use CxxWrap 0.10.

And maybe other tweaks.

**Comment by fingolfin** [comment]
This seems to fix the crashes with Julia nightly! The Travis tests with Julia nightly still error out, though, due to https://github.com/Nemocas/Nemo.jl/issues/834

**Comment by fingolfin** [comment]
This seems to work fine, and in fact fixes random crashes we get on master with Julia 1.4 and CxxWrap 0.9.1.

So, can we merge it? Then afterwards, I'd coordinate with Polymake.jl (see also https://github.com/oscar-system/Polymake.jl/issues/262) to make release of both, and update Oscar.jl

**Comment by fieker** [comment]
On Mon, Apr 20, 2020 at 09:27:20AM -0700, Max Horn wrote:
> This seems to work fine, and in fact fixes random crashes we get on master with Julia 1.4 and CxxWrap 0.9.1.
> 
> So, can we merge it? Then afterwards, I'd coordinate with Polymake.jl (see also https://github.com/oscar-system/Polymake.jl/issues/262) to make release of both, and update Oscar.jl
> 
I'd merge

File: src/LibSingular.jl [hunk 2]
```diff
@@ -2,7 +2,13 @@ module libSingular
 
 import Libdl
 using CxxWrap
-@wrapmodule(realpath(joinpath(@__DIR__, "..", "deps", "usr", "lib", "libsingularwrap." * Libdl.dlext)))
+
+const libsingularwrap_path = joinpath(@__DIR__, "..", "deps", "usr",
+                "lib", "libsingularwrap." * Libdl.dlext)
+if !isfile(libsingularwrap_path)
+    error("""Singular.jl needs to be compiled; please run `using Pkg; Pkg.build("Polymake")`""")
```
Comment (hannes14) [inline src/LibSingular.jl:9]: the error message should probably be: Pkg.build("Singular")

Comment (fingolfin) [reply src/LibSingular.jl]: Oops, good catch -- fixed now

**Comment by fingolfin** [comment]
As Nemocas/Nemo.jl#834 was fixed, all builds, including `nightly`, now pass. Nice. I'll leave the "nightly" at "allowed to fail" for now, though.

**Comment by fieker** [comment]
On Tue, Apr 21, 2020 at 04:08:31AM -0700, Max Horn wrote:
> Merged #213 into master.

Congrats.

**Pull request merged**